crates/bitwarden-vault/src/cipher/cipher.rs

```diff
@@ -456,7 +460,8 @@ impl CipherView {
         ctx: &mut KeyStoreContext<KeyIds>,
         key: SymmetricKeyId,
     ) -> Result<(), CryptoError> {
-        let old_ciphers_key = Cipher::decrypt_cipher_key(ctx, key, &self.key)?;
+        let old_ciphers_key =
+            Cipher::decrypt_cipher_key(ctx, key, &self.key.clone().map(|k| k.into()))?;
```

Comment:
Is there a representation that allows us to do this better, especially without the clone? (Could use some advice from someone with more Rust experience here)

Reply:
In this specific case, because we're setting the key immediately after and we take the cipher view by mutable reference, you can change `clone()` by `take()`, this will take the option by value leaving a `None` behind, which avoids the clone:
```suggestion
            Cipher::decrypt_cipher_key(ctx, key, &self.key.take().map(Into::into))?;
```
This won't work for the general case where we don't want to mutate the Option though.

If this is the pattern we want to go with, I feel like what we want to do is change the `key` field in Cipher from an `EncString` to a `WrappedSymmetricKey`, which would save us from doing the conversion in the first place.

Reply:
> If this is the pattern we want to go with, I feel like what we want to do is change the key field in Cipher from an EncString to a WrappedSymmetricKey, which would save us from doing the conversion in the first place.

Do we need to consider the uniffi bindings / wasm bindings here? I see they are exposed, so would this be a breaking change?

Reply:
I think for Uniffi and WASM `EncString` is basically a type alias to string, so assuming we do the same thing with `WrappedSymmetricKey` (implement Serialize/Deserialize and do the uniffi type conversion), it should basically work.

Reply:
Should be converted for uniffi and wasm now, and has been replaced in e.g. cipher and attachment.